src/Features/LanguageServer/Protocol/Handler/Diagnostics/DocumentPullDiagnosticHandler.cs

```diff
@@ -65,9 +68,9 @@ protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagno
             return ValueTaskFactory.FromResult(GetDiagnosticSources(context));
         }
 
-        protected override VSInternalDiagnosticReport[]? CreateReturn(BufferedProgress<VSInternalDiagnosticReport> progress)
+        protected override VSInternalDiagnosticReport[]? CreateReturn(BufferedProgress<VSInternalDiagnosticReport[]> progress)
         {
-            return progress.GetValues();
+            return progress.GetFlattenedValues();
```

Comment:
what's our philosophy on return vs progress?  i'd be ok only supporting IProgress if we know all our hosts support it.

Reply:
I think I'd prefer to be LSP spec compliant when it is easy (it's not too hard here).  And there's nothing spec'd that allows a server to say it *only* supports progress.